### PR TITLE
build 3.7.5 on conda-forge

### PR DESCRIPTION
**Pull request by wolfv**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
Why?

**Comment by ocefpaf** [comment]
> Why?

I'm not against it but also curious. Why?

**Comment by wolfv** [comment]
Our friends at Robocorp (who are adopting conda-forge) need this specific version currently (and for some reason it's the missing one from conda-forge :)

Unfortunately there is some weird stuff happening with line endings and the newer conda-build.

**Comment by isuruf** [comment]
That's not a reason. Please give a valid reason.

**Comment by wolfv** [comment]
They have a pinning on this specific version of the Python interpreter that they are currently getting from the defaults channel. They want to further adopt conda-forge (lot's of robot-framework packages are already on conda-forge). But if they remove defaults, the current pinning isn't working anymore, so it would be great if conda-forge had a build of 3.7.5.

Basically, it's not possible for them to remove the pinning of 3.7.5 at this point (the pinning was introduced because of a bug in 3.7.6).

Since we advise on using either strict-channel-priority or not mixing defaults I think this is the way to go.

**Comment by isuruf** [comment]
> Basically, it's not possible for them to remove the pinning of 3.7.5 at this point

Why?

**Comment by isuruf** [comment]
Is this bug fixed in 3.7.7 or later? Do you have a link to the bug report?

**Comment by kariharju** [comment]
Yeah so maybe good for me to chime on on the why questions on behalf of Robocorp.

We have at this point hundreds of user made robots running in production clouds and environments.
We are in the process of moving all these to use conda-forge and micromamba instead of anaconda defaults channel and miniconda. And on the tooling side we are moving on top of xeus-kernel. Both of these changes will free us from the locked python version, but...
Unfortunately when we started a year back we set the python version to 3.7.5 (our old kernel component also locked us to this version) so we had to put that in the templates and examples as well.. so it got pretty much everywhere.

So basically without 3.7.5 version being available in conda-forge we would have had to either keep using defaults channel or force the end-users to update all their setups. With 3.7.5 we can get the migrations rolling without forcing get more people to conda-forge.

File: recipe/patches/0016-Unvendor-openssl.patch [hunk 1]
```diff
@@ -1,22 +1,8 @@
...
-
----
- PCbuild/_ssl.vcxproj         |  3 ---
- PCbuild/_ssl.vcxproj.filters |  5 +----
- PCbuild/openssl.props        | 21 ++-------------------
- PCbuild/python.props         |  6 +++---
- PCbuild/python.vcxproj       |  3 +++
- PCbuild/pythonw.vcxproj      |  3 +++
- 6 files changed, 12 insertions(+), 29 deletions(-)
-
```
Comment (isuruf) [inline recipe/patches/0016-Unvendor-openssl.patch:14]: Please keep this as it is

File: recipe/patches/0016-Unvendor-openssl.patch [hunk 2]
```diff
@@ -25,7 +11,12 @@ index 506d5ca..ba955c1 100644
...
    </ItemGroup>
    <ItemGroup>
-     <ResourceCompile Include="..\PC\python_nt.rc" />
+-    <ResourceCompile Include="..\PC\python_nt.rc" />
+-  </ItemGroup>
++    <ResourceCompile Include="..\PC\python_nt.rc" />
  </ItemGroup>
+   <ItemGroup>
+     <ProjectReference Include="pythoncore.vcxproj">
+       <Project>{cf7ac3d1-e2df-41d2-bea6-1e2556cdea26}</Project>
```
Comment (isuruf) [inline recipe/patches/0016-Unvendor-openssl.patch:19]: Please revert

File: recipe/patches/0016-Unvendor-openssl.patch [hunk 3]
```diff
@@ -78,27 +69,32 @@ index b43c674..16d00fc 100644
...
 +    <opensslDir>$(OPENSSL_DIR)\</opensslDir>
 +    <opensslOutDir>$(opensslDir)bin</opensslOutDir>
 +    <opensslIncludeDir>$(opensslDir)include</opensslIncludeDir>
      <nasmDir>$(ExternalsDir)\nasm-2.11.06\</nasmDir>
      <zlibDir>$(ExternalsDir)\zlib-1.2.11\</zlibDir>
+-    
+-    <!-- Suffix for all binaries when building for debug -->
++    
    <!-- Suffix for all binaries when building for debug -->
+     <PyDebugExt Condition="'$(PyDebugExt)' == '' and $(Configuration) == 'Debug'">_d</PyDebugExt>
```
Comment (isuruf) [inline recipe/patches/0016-Unvendor-openssl.patch:90]: Please revert

File: recipe/patches/0016-Unvendor-openssl.patch [hunk 4]
```diff
@@ -121,7 +122,9 @@ index 69c5957..faff342 100644
...
--- 
-2.20.1
-
+-    <ProjectReference Include="pythoncore.vcxproj">
+-      <Project>{cf7ac3d1-e2df-41d2-bea6-1e2556cdea26}</Project>
++    <ProjectReference Include="pythoncore.vcxproj">
      <Project>{cf7ac3d1-e2df-41d2-bea6-1e2556cdea26}</Project>
+       <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+     </ProjectReference>
+   </ItemGroup>
```
Comment (isuruf) [inline recipe/patches/0016-Unvendor-openssl.patch:130]: Please revert

File: recipe/patches/0017-Unvendor-sqlite3.patch [hunk 5]
```diff
@@ -1,15 +1,3 @@
...
-Date: Thu, 24 Oct 2019 21:55:24 -0500
-Subject: [PATCH 17/20] Unvendor sqlite3
-
----
- PCbuild/_sqlite3.vcxproj |  9 ++++-----
- PCbuild/pcbuild.sln      |  2 --
- PCbuild/python.props     |  2 +-
- PCbuild/sqlite3.vcxproj  | 12 ++++++------
- 4 files changed, 11 insertions(+), 14 deletions(-)
-
```
Comment (isuruf) [inline recipe/patches/0017-Unvendor-sqlite3.patch:12]: Please revert

File: recipe/patches/0017-Unvendor-sqlite3.patch [hunk 6]
```diff
@@ -53,18 +41,28 @@ index 554bd13..ef25b2c 100644
...
+     <opensslIncludeDir>$(opensslDir)include</opensslIncludeDir>
+     <nasmDir>$(ExternalsDir)\nasm-2.11.06\</nasmDir>
+     <zlibDir>$(ExternalsDir)\zlib-1.2.11\</zlibDir>
+-    
    <!-- Suffix for all binaries when building for debug -->
++    
++    <!-- Suffix for all binaries when building for debug -->
+     <PyDebugExt Condition="'$(PyDebugExt)' == '' and $(Configuration) == 'Debug'">_d</PyDebugExt>
+     
+     <!-- Suffix for versions/keys when building with test markers -->
```
Comment (isuruf) [inline recipe/patches/0017-Unvendor-sqlite3.patch:65]: Please revert

**Review by isuruf — CHANGES_REQUESTED** [review]

**Comment by isuruf** [comment]
That makes more sense than what @wolfv did. Please add yourselves as maintainers here.

**Comment by wolfv** [comment]
@isuruf the line ending behavior in conda-build has changed at some point, that's why the original patches don't apply cleanly anymore. Do you have some advice how to create clean patches?
I tried with `git apply old_patch.patch --ignore-whitespace` and then creating a new patch from that...

**Comment by isuruf** [comment]
Okay then, please restore the metadata (Author, email, etc) and add yourselves as maintainers

**Comment by wolfv** [comment]
Thanks, will do

**Comment by isuruf** [comment]
@wolfv, can you send a PR to update this branch to 3.7.9 again?

**Comment by isuruf** [comment]
@wolfv, ping on this

**Comment by wolfv** [comment]
How do I do it in such a way that no new 3.7.9 packages are uploaded to CF?

**Comment by isuruf** [comment]
That's fine. We also need to update the build number to something like 100 to avoid a conda solver issue.

**Comment by isuruf** [comment]
Any updates?

**Comment by isuruf** [comment]
Ping again

**Comment by wolfv** [comment]
PR is open

**Comment by wolfv** [comment]
https://github.com/conda-forge/python-feedstock/pull/446
